hugr-core/src/import.rs

```diff
@@ -216,24 +216,26 @@ impl<'a> Context<'a> {
                 }
             }
 
-            if inputs.is_empty() || outputs.is_empty() {
-                return Err(error_unsupported!(
-                    "link {}#{} is missing either an input or an output port",
-                    link_id.0,
-                    link_id.1
-                ));
-            }
-
-            // We connect the first output to all the inputs, and the first input to all the outputs
-            // (except the first one, which we already connected to the first input). This should
-            // result in the hugr having a (hyper)edge that connects all the ports.
-            // There should be a better way to do this.
-            for (node, port) in inputs.iter() {
-                self.hugr.connect(outputs[0].0, outputs[0].1, *node, *port);
-            }
-
-            for (node, port) in outputs.iter().skip(1) {
-                self.hugr.connect(*node, *port, inputs[0].0, inputs[0].1);
+            match (inputs.as_slice(), outputs.as_slice()) {
+                ([], []) => {
+                    unreachable!();
+                }
+                (_, [output]) => {
+                    for (node, port) in inputs.iter() {
+                        self.hugr.connect(output.0, output.1, *node, *port);
+                    }
+                }
+                ([input], _) => {
```

Comment:
not great to have these untested lines especially since they are adjacent to a bug, I think benefits from a dedicated test

Reply:
Covered now.